full-service/src/json_rpc/wallet.rs

```diff
@@ -0,0 +1,194 @@
+// Copyright (c) 2020-2021 MobileCoin Inc.
+
+//! Entrypoint for Wallet API.
+
+use crate::{
+    db,
+    db::account::AccountID,
+    json_rpc,
+    json_rpc::{
+        api_v1::wallet_api::{help_str_v1, wallet_api_inner_v1, JsonCommandRequestV1},
+        json_rpc_request::{help_str_v2, JsonCommandRequest, JsonCommandRequestV2},
+        json_rpc_response::{
+            format_error, JsonCommandResponse, JsonCommandResponseV2, JsonRPCResponse,
+        },
+    },
+    service::{account::AccountService, WalletService},
+};
+use mc_connection::{
+    BlockchainConnection, HardcodedCredentialsProvider, ThickClient, UserTxConnection,
+};
+use mc_fog_report_validation::{FogPubkeyResolver, FogResolver};
+use rocket::{get, post, routes};
+use rocket_contrib::json::Json;
+use serde_json::Map;
+use std::{convert::TryFrom, iter::FromIterator};
+
+/// State managed by rocket.
+pub struct WalletState<
+    T: BlockchainConnection + UserTxConnection + 'static,
+    FPR: FogPubkeyResolver + Send + Sync + 'static,
+> {
+    /// The Wallet Service implementation.
+    pub service: WalletService<T, FPR>,
+}
+
+/// The route for the Full Service Wallet API.
+#[post("/wallet", format = "json", data = "<command>")]
```

Comment:
It feels weird to me that this handles requests to both (and eventually, all?) API versions. It seems like that adds boilerplate (i.e. the if / else blocks, an api_version field to the Requests, etc.), but mostly it seems to couple together the code paths of each API version, when ideally those are as independent as possible.

My vote would be for a top-level module separation (something like json_rpc/v1 and json_rpc/v2), with routes mapped under the top-level URLs /v1/foo and /v2/foo, etc.